dimod/discrete/discrete_quadratic_model.py

```diff
@@ -217,6 +216,150 @@ def add_linear_equality_constraint(self, terms: LinearTriplets,
         self._cydqm.add_linear_equality_constraint(
             index_terms, lagrange_multiplier, constant)
 
+    def add_linear_inequality_constraint(self, terms: LinearTriplets,
+                                         lagrange_multiplier: float,
+                                         label: str,
+                                         constant: int = 0,
+                                         lb: int = np.iinfo(np.int64).min,
+                                         ub: int = 0,
+                                         slack_method: str = "log2",
+                                         cross_zero: bool = False)\
+            -> LinearTriplets:
+
+        """Add a linear inequality constraint as a quadratic objective.
+
+        Adds a linear inequality constraint of the form:
+
+        math:'lb <= \sum_{i,k} a_{i,k} x_{i,k} + constant <= ub'
+        to the discrete quadratic model as a quadratic objective.
+        Coefficients should be integers.
+        For constraints with fractional coefficients, multiply both sides of
+        the inequality by an appropriate factor of ten to attain or approximate
+        integer coefficients.
+
+        Args:
+            terms:
+                A list of tuples of the type (variable, case, bias).
+                Each tuple is evaluated to the term (bias * variable_case).
+                All terms in the list are summed.
+            lagrange_multiplier:
+                A weight or the penalty strength. This value is multiplied by
+                the entire constraint objective and added to the
+                discrete quadratic model (it doesn't appear explicitly in the
++               equation above).
+            label:
+                Prefix used to label the slack variables used to create the new
+                objective.
+            constant:
+                The constant value of the constraint.
+            lb:
+                lower bound for the constraint
+            ub:
+                upper bound for the constraint
+            slack_method:
+                "The method for adding slack variables. Supported methods are:
+                - log2: Adds up to log2(ub - lb) number of dqm variables each
+                        with two cases to the constraint.
+                - log10: Adds log10 dqm variables each with up to 10 cases.
+            cross_zero:
+                 When True, adds zero to the domain of constraint
+
+        Returns:
+            slack_terms:  A list of tuples of the type (variable, case, bias)
+                for the new slack variables.
+                Each tuple is evaluated to the term (bias * variable_case).
+                All terms in the list are summed.
+
+       """
+
+        if slack_method not in ['log2', 'log10']:
+            raise ValueError(
+                r"expected slack_method to be 'log2' or 'log10', "
```

Comment:
don't need the `r` on the first line, and I think you mean `f` on the second one.